forge-gui/res/cardsfolder/upcoming/kaito_dancing_shadow.txt

```diff
@@ -0,0 +1,15 @@
+Name:Kaito, Dancing Shadow
+ManaCost:2 U B
+Types:Legendary Planeswalker Kaito
+Loyalty:3
+T:Mode$ DamageDoneOnce | CombatDamage$ True | ValidSource$ Creature.YouCtrl | TriggerZones$ Battlefield | ValidTarget$ Player | Execute$ TrigChoose | TriggerDescription$ Whenever one or more creatures you control deal combat damage to a player, you may return one of them to its owner's hand. If you do, you may activate loyalty abilities of NICKNAME twice this turn rather than only once.
+SVar:TrigChoose:DB$ ChooseCard | DefinedCards$ TriggeredSources | SubAbility$ DBBounce
```

Comment:
Would be better to just use `Cost$ Return<1/Card.TriggeredSources>`

Reply:
Seems like a cleaner way to do it. I'm not quite sure where this clause would go though. I tried attaching the cost to the DBLoyalty Effect, but no prompt to bounce appears. Where should this go?

Reply:
Yep, you want the Effect line – just change it from `DB$` to `AB$` and it will handle the Cost properly

Reply:
Did you make it the new root? `SVar:TrigLoyalty:AB$ Effect` should work

Reply:
Just tested this a moment ago. The triggered ability to bounce is happening, but choosing 'yes' causes it to resolve abruptly, nothing is returned, and Kaito can't be activated twice. I checked that the root is correctly executing "TrigLoyalty" from DamageDoneOnce. Here is the new TrigLoyalty line:

`SVar:TrigLoyalty:AB$ Effect | Cost$ Return<1/Card.TriggeredSources> | StaticAbilities$ PWTwice | ConditionDefined$ Remembered | ConditionPresent$ Card | SubAbility$ DBCleanup | SpellDescription$ You may activate the loyalty abilities of NICKNAME twice this turn rather than only once.`

Thanks for the help all!

Reply:
My gut says the syntax on `Return<1/Card.TriggeredSources>` isn't perfect and that's what's breaking it

Reply:
Why would you keep the conditions when you no longer remember anything? ;)

Reply:
> Why would you keep the conditions when you no longer remember anything? ;)

Ah even better 🐛 🥾

Reply:
Now it reads:

`SVar:TrigLoyalty:AB$ Effect | Cost$ Return<1/Card.TriggeredSources> | StaticAbilities$ PWTwice | SpellDescription$ You may activate the loyalty abilities of NICKNAME twice this turn rather than only once.`

Good shout on removing the remembered conditions, since we won't be using them. But the issue remains where I cannot select a creature to bounce. Do we think it is the Card.TriggeredSources that needs changing?

Reply:
Are you on the discord in general or our discord in particular?

https://discord.gg/wXRXAHwx

Reply:
I am going to debug this a bit

Reply:
Yes, I am Suthro :) I can DM you my working script on discord if you need

Reply:
Also note that the Execute and any SubAbilities following a trigger don't/need use `SpellDescription`

All you need is the `Description` on the static line (NumLoyaltyAct)

Reply:
Okay, with the engine change accommodation, the updated Kaito script seems to work fine. I'll keep this convo open briefly just in case there are any cleanup revisions to be made. Thanks for the help all!